Python/ast.c

```diff
@@ -584,6 +584,671 @@ PyAST_Validate(mod_ty mod)
     return res;
 }
 
+char *
+wrap_parens(char * str)
+{
+    char *result;
+    result = PyMem_RawMalloc(strlen(str) + 3);
```

Comment:
Results of all allocations should be checked for NULL. I think it is worth to implement simple accumulator that uses overallocated array and makes concatenations for linear time.